subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBinary.java

```diff
@@ -150,4 +174,73 @@ public FileCollection getObjects() {
     public Property<SwiftCompile> getCompileTask() {
         return compileTaskProperty;
     }
+
+    @Inject
+    protected NativeDependencyCache getNativeDependencyCache() {
+        throw new UnsupportedOperationException();
+    }
+
+    private class ModulePath implements MinimalFileSet, Buildable {
+        private final Configuration importPathConfig;
+
+        private Set<File> result;
+
+        ModulePath(Configuration importPathConfig) {
+            this.importPathConfig = importPathConfig;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Module include path for " + DefaultSwiftBinary.this.toString();
+        }
+
+        @Override
+        public Set<File> getFiles() {
+            if (result == null) {
+                result = Sets.newLinkedHashSet();
+                Map<String, ModuleMap> moduleMaps = Maps.newHashMap();
+                for (ResolvedArtifactResult artifact : importPathConfig.getIncoming().getArtifacts()) {
+                    Usage usage = artifact.getVariant().getAttributes().getAttribute(Usage.USAGE_ATTRIBUTE);
+                    if (usage != null && Usage.C_PLUS_PLUS_API.equals(usage.getName())) {
+                        String moduleName;
+
+                        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
+                        if (ModuleComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ModuleComponentIdentifier) id).getModule();
+                        } else if (ProjectComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ProjectComponentIdentifier) id).getProjectName();
+                        } else if (LibraryBinaryIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((LibraryBinaryIdentifier) id).getLibraryName();
+                        } else {
+                            throw new IllegalArgumentException("Could not determine the name of " + id.getDisplayName() + ": unknown component identifier type: " + id.getClass().getSimpleName());
+                        }
+
+                        ModuleMap moduleMap;
+                        if (moduleMaps.containsKey(moduleName)) {
+                            moduleMap = moduleMaps.get(moduleName);
+                        } else {
+                            moduleMap = new ModuleMap(moduleName, Lists.<String>newArrayList());
+                            moduleMaps.put(moduleName, moduleMap);
```

Comment:
This seems a little hokey (name collisions above and ordering issues below). Is there a way to get the results grouped by their component id so we could do the conversion all at once?

Reply:
I don't know of some way to say "give me all of the artifacts for dependency X", but I've changed the map so that it's keyed with the component id, which should avoid collisions and allow different maps to be generated for different components.  I've also changed it so order is preserved (at least in as much as the order of components appear in the artifact results).